tardis/adapters/batchsystems/htcondor.py

```diff
@@ -47,9 +47,9 @@ async def htcondor_status_updater(
 
     try:
         logger.debug(f"HTCondor status update is running. Command: {cmd}")
-        condor_status = await async_run_command(cmd)
+        condor_status = await executor.run_command(cmd)
```

Comment:
I followed the implementation in the HTCondor site adapter: https://github.com/MatterMiners/tardis/blob/b3ea6fa926048c7812632a43663f0a5b005ee443/tardis/adapters/sites/htcondor.py#L32
Is there a reason why `async_run_command` is not used there?

Reply:
No, there is no reason. Probably the best way to solve it by removing `async_run_command` and implementing the workaround in the `ShellExecutor`. Most like nobody run into this issue using the HTCondor site adapter yet, because either modern Python versions (>3.7) are used or the `SSHExecutor` is used.

Reply:
If this workaround only affects `ShellExecutor`, then I also think this would be the best idea. If it is fine for you, I could do this as part of this PR or as part of a new one. Might take a while though because I'll probably be fighting the tests again.

Reply:
Yes, it only affects affects the `ShellExecutor`. Would be awesome if you could do that in this pull request.